localstack/utils/net.py

```diff
@@ -289,3 +291,48 @@ def _try_reserve_port(self, port: IntOrPort, duration: int) -> int:
             return port.port
         else:
             raise PortNotAvailableException(f"The given port ({port}) is already reserved.")
+
+
+@singleton_factory
+def get_docker_host_from_container() -> str:
+    """
+    Get the hostname/IP to connect to the host from within a Docker container (e.g., Lambda function).
+    The logic is roughly as follows:
+      1. return `host.docker.internal` if we're running in host mode, in a non-Linux OS
+      2. return the IP address that `host.docker.internal` (or alternatively `host.containers.internal`)
+        resolves to, if we're inside Docker
+      3. return the Docker bridge IP (config.DOCKER_BRIDGE_IP) as a fallback, if option (2) fails
+    """
+    result = config.DOCKER_BRIDGE_IP
+    try:
+        if not config.is_in_docker and not config.is_in_linux:
+            # If we're running outside Docker (in host mode), and would like the Lambda containers to be able
+            # to access services running on the local machine, return `host.docker.internal` accordingly
+            if config.LOCALSTACK_HOSTNAME == constants.LOCALHOST:
+                result = "host.docker.internal"
+        # update LOCALSTACK_HOSTNAME if host.docker.internal is available
+        if config.is_in_docker:
+            try:
+                result = socket.gethostbyname("host.docker.internal")
+            except socket.error:
+                result = socket.gethostbyname("host.containers.internal")
+            # TODO still required? - remove
+            # if config.LOCALSTACK_HOSTNAME == config.DOCKER_BRIDGE_IP:
```

Comment:
I _think_ this it is no longer needed and we can remove it, but would be good to also get a quick pair of eyes from @simonrw 👀